internal/tofu/node_resource_plan.go

```diff
@@ -438,3 +450,15 @@ func (n *nodeExpandPlannableResource) resourceInstanceSubgraph(ctx context.Conte
 	graph, graphDiags := b.Build(ctx, addr.Module)
 	return graph, diags.Append(graphDiags).ErrWithWarnings()
 }
+
+// Close implements closableResource
+func (n *nodeExpandPlannableResource) Close() (diags tfdiags.Diagnostics) {
+	if n.Addr.Resource.Mode != addrs.EphemeralResourceMode {
+		return diags
+	}
+
+	for _, c := range n.closers {
+		diags = diags.Append(c())
```

Comment:
Should we consider doing this in parallel? If each close operation takes multiple seconds (worst case), this could balloon out pretty quickly.